README.md

```diff
@@ -12,3 +12,205 @@ The tool contains a number of commands. Currently, these are:
+```
+### Ubuntu
+
+To install a `dnsi` package, you need the 64-bit version of one of these
+Ubuntu versions:
+
+- Ubuntu Jammy 22.04 (LTS)
+- Ubuntu Focal 20.04 (LTS)
+- Ubuntu Bionic 18.04 (LTS)
+
+Packages are available for the `amd64`/`x86_64` architecture only.
+
+First update the `apt` package index: 
+
+``` bash 
+sudo apt update
+```
+
+Then install packages to allow `apt` to use a repository over HTTPS:
+
+``` bash
+sudo apt install \
+  ca-certificates \
+  curl \
+  gnupg \
+  lsb-release
+```
+
+Add the GPG key from NLnet Labs:
+
+``` bash
+curl -fsSL https://packages.nlnetlabs.nl/aptkey.asc | sudo gpg --dearmor -o /usr/share/keyrings/nlnetlabs-archive-keyring.gpg
+```
+
+Now, use the following command to set up the *main* repository:
+
+``` bash
+echo \
+"deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/nlnetlabs-archive-keyring.gpg] https://packages.nlnetlabs.nl/linux/ubuntu \
+$(lsb_release -cs) main" | sudo tee /etc/apt/sources.list.d/nlnetlabs.list > /dev/null
+```
+
+Update the `apt` package index once more:
+
+``` bash
+sudo apt update
+```
+
+You can now install `dnsi` with:
+
+``` bash
+sudo apt install dnsi
+```
+
+### RHEL and compatible systems
+
+To install a `dnsi` package, you need Red Hat Enterprise Linux (RHEL) 7,
+8 or 9, or compatible operating system such as Rocky Linux. Packages are
+available for the `amd64`/`x86_64` architecture only.
+
+First create a file named `/etc/yum.repos.d/nlnetlabs.repo`, enter this
+configuration and save it:
+
+``` text
+[nlnetlabs]
+name=NLnet Labs
+baseurl=https://packages.nlnetlabs.nl/linux/centos/$releasever/main/$basearch
+enabled=1
+```
+
+Add the GPG key from NLnet Labs:
+
+``` bash
+sudo rpm --import https://packages.nlnetlabs.nl/aptkey.asc
+```
+
+You can now install `dnsi` with:
+
+``` bash
+sudo yum install -y dnsi
+```
+
+## Building
+
+`dnsi` is written in Rust. The Rust compiler runs on, and compiles to, a
+great number of platforms, though not all of them are equally supported. The
+official [Rust Platform
+Support](https://doc.rust-lang.org/nightly/rustc/platform-support.html) page
+provides an overview of the various support levels.
+
+### Installing Rust
+
+While some system distributions include Rust as system packages, `dnsi`
+relies on a relatively new version of Rust, currently 1.74 or newer.
+We therefore suggest to use the canonical Rust installation via a tool called
+`rustup`.
+
+Assuming you already have `curl` installed, you can install `rustup` and Rust
+by simply entering:
+
+``` bash
+curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
+```
+
+Alternatively, visit the [Rust website](https://www.rust-lang.org/tools/install) for other installation methods. Also refer to this page for notes
+on updating Rust, and configuring the `PATH` environment variable.
+
+### Installing and updating `dnsi`
+
+After successfully installing Rust, installing `dnsi` is as simple as
+entering:
+
+``` bash
+cargo install dnsi
```

Comment:
```suggestion
cargo install --locked dnsi
```
I think this helps a bit if some dependency breaks semver. It's a good extra guard to have, I think.

Reply:
The downside of using locked ia that installers won't benefit from supposedly compatible fixes to dependencies used by dnsi.

Reply:
One other reason to use locked is that is how Ploutos builds binaries for inclusion in packages and thus anyone installing using locked gets the same versions as if they install from a package built by us.